gui/qt/__init__.py

```diff
@@ -142,6 +143,7 @@ def __init__(self, config, daemon, plugins):
         if self.has_auto_update_check():
             self._start_auto_update_timer(first_run = True)
         self.app.focusChanged.connect(self.on_focus_change)  # track last window the user interacted with
+        self.shutdown_signal.connect(self.close, Qt.QueuedConnection)
```

Comment:
I wonder if the default of `Qt.AutoConnection` would be sufficient here -- since that does direct call if same thread, or queued call if other thread.

Hmm.  Probably the nature of signal handlers (they can arrive at any time) makes Queued Connection more appropriate.   Yeah.  Good **call**.